Gems/Atom/Feature/Common/Assets/Materials/Pipelines/LowEndPipeline/LowEndPipelineScript.lua

```diff
@@ -0,0 +1,39 @@
+--------------------------------------------------------------------------------------
+--
+-- Copyright (c) Contributors to the Open 3D Engine Project.
+-- For complete copyright and license terms please see the LICENSE at the root of this distribution.
+--
+-- SPDX-License-Identifier: Apache-2.0 OR MIT
+--
+--
+--
+----------------------------------------------------------------------------------------------------
+
+function MaterialTypeSetup(context)
+    lightingModel = context:GetLightingModelName()
+    Print('Material type uses lighting model "' .. lightingModel .. '".')
+
+    context:ExcludeAllShaders()
+    
+    if(lightingModel == "Base") then
+        context:IncludeShader("DepthPass")
+        context:IncludeShader("ShadowmapPass")
+        context:IncludeShader("ForwardPass_BaseLighting")
+        return true
+    end
+
+    if(lightingModel == "Standard" or lightingModel == "Enhanced") then
+        if(lightingModel == "Enhanced") then
+            Warning("The low end pipeline does not support the Enhanced lighting model. Will use Standard lighting as a fallback.")
+        end
+
+        context:IncludeShader("DepthPass")
+        context:IncludeShader("ShadowmapPass")
+        context:IncludeShader("ForwardPass_StandardLighting")
+        return true
+    end
+    
+    Error('Unsupported lighting model "' .. lightingModel .. '".')
+    return false
+end
```

Comment:
What other use cases do you see for requiring lua? This is cool and potentially powerful but my concern is that it means somebody has to write lua code. I have not looked through all of the lua files in the PR yet. So far, it kind of looks like inclusion and exclusion lists, basic rules that could be set up using a table in a reflected data structure instead of script. Doing it that way means it can be configured from one of our tools or Python scripts.

Reply:
The Material Pipeline section of the RFC goes into those details: https://github.com/o3de/sig-graphics-audio/blob/main/rfcs/MaterialPipelineAbstraction.md#material-pipeline

There could be other settings besides lighting model that dictate which shaders should be compiled. You could also use the script to set preprocessor flags (though I don't think that was covered in the RFC). Not every material pipeline will necessarily use these features, but I think supporting a script is the most flexible way to enable use cases that we haven't thought of yet. The lua API can be easily extended to support new things, like configuring depth/stencil states.